daphne/src/lib.rs

```diff
@@ -95,6 +95,12 @@ impl From<::hpke::HpkeError> for DapError {
     }
 }
 
+impl From<TransitionFailure> for DapError {
```

Comment:
I think the code would be clearer without this. Beware of overusing the `From<A> for B` -- in general it's best to handle things explicitly.

Reply:
I thought the code would look cleaner because we just have to use `?`, but I have removed this and reverted back to handling the error explicitly.